adam-core/src/test/scala/org/bdgenomics/adam/rdd/SortedGenomicRDDSuite.scala

```diff
@@ -0,0 +1,310 @@
+            .build()
+        )
+        .setSampleId("3")
+        .build()
+    }
+
+    genotypeRddBuilder += {
+      Genotype.newBuilder()
+        .setContigName("chr1")
+        .setStart(8L)
+        .setEnd(12L)
+        .setVariant(
+          Variant.newBuilder()
+            .setStart(8L)
+            .setEnd(12L)
+            .setAlternateAllele("A")
+            .setReferenceAllele("T")
+            .build()
+        )
+        .setSampleId("3")
+        .build()
+    }
+
+    val featureRddBuilder = new ListBuffer[Feature]()
+
+    featureRddBuilder += {
+      Feature.newBuilder()
+        .setContigName("chr1")
+        .setStart(61L)
+        .setEnd(62L)
+        .build()
+    }
+
+    featureRddBuilder += {
+      Feature.newBuilder()
+        .setContigName("chr1")
+        .setStart(11L)
+        .setEnd(15L)
+        .build()
+    }
+
+    featureRddBuilder += {
+      Feature.newBuilder()
+        .setContigName("chr1")
+        .setStart(3L)
+        .setEnd(6L)
+        .build()
+    }
+
+    featureRddBuilder += {
+      Feature.newBuilder()
+        .setContigName("chr1")
+        .setStart(6L)
+        .setEnd(8L)
+        .build()
+    }
+
+    featureRddBuilder += {
+      Feature.newBuilder()
+        .setContigName("chr1")
+        .setStart(50L)
+        .setEnd(52L)
+        .build()
+    }
+
+    featureRddBuilder += {
+      Feature.newBuilder()
+        .setContigName("chr1")
+        .setStart(1L)
+        .setEnd(2L)
+        .build()
+    }
+
+    val genotypes = GenotypeRDD(sc.parallelize(genotypeRddBuilder), sd, Seq()).repartitionAndSort(2)
+    genotypes.rdd.mapPartitionsWithIndex((idx, iter) => {
+      iter.map(f => (idx, f))
+    }).collect
+    val features = FeatureRDD(sc.parallelize(featureRddBuilder), sd)
+    val x = features.copartitionByReferenceRegion(genotypes)
+    val z = x.rdd.mapPartitionsWithIndex((idx, iter) => {
+      if (idx == 0 && iter.size != 6) {
+        Iterator(true)
+      } else if (idx == 1 && iter.size != 2) {
+        Iterator(true)
+      } else {
+        Iterator()
+      }
+    })
+
+    x.rdd.mapPartitionsWithIndex((idx, iter) => {
+      iter.map(f => (idx, f))
+    }).collect
+    assert(z.collect.length == 0)
+
+  }
+
+  sparkTest("testing that sorted shuffleRegionJoin matches unsorted") {
+    val x = sc.loadBam(getClass.getResource("/bqsr1.sam").getFile)
+    // sort and make into 16 partitions
+    val z = x.repartitionAndSort(16)
+    // perform join using 32 partitions
+    val b = z.shuffleRegionJoin(x, Some(1))
+    val c = x.shuffleRegionJoin(z, Some(1))
+    val d = c.rdd.map(f => (f._1.getStart, f._2.getEnd)).collect.toSet
+    val e = b.rdd.map(f => (f._1.getStart, f._2.getEnd)).collect.toSet
+
+    val setDiff = d -- e
+    assert(setDiff.isEmpty)
+
+    assert(b.rdd.count == c.rdd.count)
+  }
+
+  sparkTest("testing that sorted fullOuterShuffleRegionJoin matches unsorted") {
+    val x = sc.loadBam(getClass.getResource("/bqsr1.sam").getFile)
```

Comment:
This is in a lot of places; I'm going to stop highlighting it but please fix it throughout this test suite.